### PR TITLE
ENH: Use ccache in the CircleCI build

### PR DESCRIPTION
**Pull request by thewtex**
Change-Id: I758d9812416d4df2c6153ce4a2a69142fe258c4e
Suggested-by: Bradley Lowekamp <blowekamp@mail.nih.gov>

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -8,10 +8,28 @@ jobs:
...
           name: Run CTest build and report to CDash
           no_output_timeout: 2.0h
           command: /cmd.sh
+      - run:
+          name: ccache stats
+          when: always
+          command: |
+            ccache --show-stats
+      - save_cache:
+          key: 'ccache-{{ arch }}-{{ .Branch }}'
```
Comment (thewtex) [inline .circleci/config.yml]: @blowekamp the `-{{ epoch }}` in ITK's CircleCI configuration is a bug?

Comment (blowekamp) [reply .circleci/config.yml]: If the epoch is not there then the string matches an existing entry an it is no updated.

**Comment by blowekamp** [comment]
I triggered a rebuild. The ccache appeared empty.

The compiler CMake is picking up is "/usr/bin/c++" vs "/usr/lib/ccache/cc" on the ITK CircleCI build.

Is ccache installed on the docker image?

**Comment by thewtex** [comment]
ccache is in the docker image, but there was a bug passing it in the build via `CMAKE_CXX_COMPILER_LAUNCHER`. Addressing that...

File: .circleci/config.yml (hunk 1 above)
Comment (thewtex) [reply .circleci/config.yml]: Ok, thanks for the explanation.

**Comment by blowekamp** [comment]
CircleCI cache is a different path that what ccache is using:
```
#!/bin/bash -eo pipefail
ccache --show-stats
cache directory                     /root/.ccache
primary config                      /root/.ccache/ccache.conf
secondary config      (readonly)    /etc/ccache.conf
cache hit (direct)                     0
cache hit (preprocessed)               0
cache miss                          4374
cache hit rate                      0.00 %
cleanups performed                     4
files in cache                     10937
cache size                           1.7 GB
max cache size                       2.0 GB
```
vs
```
Creating cache archive...
Uploading cache archive...
Stored Cache to ccache-arch1-linux-amd64-6_62-pull/40-1513957684
  * /home/circleci/.ccache
```

**Comment by thewtex** [comment]
Addressing the cache path...

**Comment by thewtex** [comment]
Some success:
```
#!/bin/bash -eo pipefail
ccache --show-stats
cache directory                     /root/.ccache
primary config                      /root/.ccache/ccache.conf
secondary config      (readonly)    /etc/ccache.conf
cache hit (direct)                  2373
cache hit (preprocessed)             750
cache miss                          1251
cache hit rate                     71.40 %
cleanups performed                    18
files in cache                     12674
cache size                           1.8 GB
max cache size                       2.0 GB
```

Reduces build time from 50 minutes to 30 minutes.

**Pull request merged**